src/ImageBackingStore.cpp

```diff
@@ -385,6 +385,19 @@ void ImageBackingStore::flush()
     }
 }
 
+bool ImageBackingStore::truncate(uint64_t size)
+{
+    if (m_iscontiguous || m_isrom || m_israw || m_isreadonly_attr)
```

Comment:
I think we could set `m_iscontiguous` to false here, so that file can be later truncated even if it was initially opened as contiguous file.

Reply:
Implemented the contiguous fallback on truncate.